core/src/main/groovy/noe/tomcat/configure/TomcatConfigurator.groovy

```diff
@@ -63,6 +63,8 @@ class TomcatConfigurator {
    * have no unique id. *
    *
    * If there is not http (non-secure) connector then new connector is created.
+   * If there is already one, the connector attributes will be updated
```

Comment:
We need to resolve this as well before merging.
I am still not sure about this one.

Imagine that you are new to `TomcatConfigurator` and you want to call code like
````
 ConnectorCertificateTomcat certificateObj = new ConnectorCertificateTomcat()
      .setCertificateFile(certFile)
      .setCertificateChainFile(chain)
      .setCertificateKeyAlias(keyAlias)
      .setCertificateKeyFile(keyFile)
      .setCertificateKeyPassword(keyPass)
      .setCertificateKeystoreFile(keystoreFile)
      .setCertificateKeystorePassword(keystorePass)
      .setCertificateKeystoreType(keystoreType)
      .setCertificateKeystoreProvider(keystoreProvider)
      .setCertificateType(certType)

    ConnectorSSLHostConfigTomcat sslHostConfObj = new ConnectorSSLHostConfigTomcat()
      .setCertificate(certificateObj)

    new TomcatConfigurator(tomcat)
      .httpsConnector(
        new SecureHttpConnectorTomcat()
          .setPort(testHttpsPort)
          .setSSLHostConfigs(sslHostConfObj)
      )
````

How you know, what configurations update the `server.xml` and what sets to specified values exactly? If you find out that some of the methods work differently than others (some sets, some updates), what would you think?

How do you suggest to handle this situation?

Reply:
Hmmm I'm understanding what you mean
Maybe I should rephrase it to something like 

   `* If there is no secure HTTP connector then new secure HTTP connector is created.`
   `* If there is already one, its new attributes will be added along with the old ones`
   `* Inner elements have to be written to their exact format as a whole`
   
what do you think?

Reply:
I am sure, that the text is good, as you suggesting. What I am not sure, that it will prevent surprises (not everybody reads the doc).
Any other idea?

Reply:
I believe that someone not reading the docs, and not seeing the code-functionality will of course lead the user to surprises and it seems legit. From our side, I believe there is nothing more we can do :/ The naming is pretty good, java doc is present and the code is very clear. Nothing else comes to my mind right now

Reply:
Reading this again after a few weeks and to be doc more clear, it would imho help the following little change:

```
* If there is no secure HTTP connector, then a new secure HTTP connector is created.
* If there is already one, its new attributes will be added along with the old ones and old ones will be updated. Attributes are configured by methods with parameter type of java standard classes and primitive types.
* Inner elements have to be written to their exact form as a whole. Inner elements are configured by methods with custom classes.
```

And BTW, happy new year, let's hope that it will be better in every aspect!

Reply:
Made the little change

Happy new year to you and your beloved ones Jan. Wishing the best :)